pipeline.cpp

```diff
@@ -80,7 +118,45 @@ void DrawLine(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* cons
  ************************************************************/
 void DrawTriangle(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* const attrs, Attributes* const uniforms, FragmentShader* const frag)
 {
-    // Your code goes here
+
+    //By finding the max and min of both are x and y axis we can find the outer box of the triangle. 
+    int maxX = fmax((int)triangle[0].x, fmax((int)triangle[1].x, (int)triangle[2].x));
+    int minX = fmin((int)triangle[0].x, fmin((int)triangle[1].x, (int)triangle[2].x));
+    int maxY = fmax((int)triangle[0].y, fmax((int)triangle[1].y, (int)triangle[2].y));
+    int minY = fmin((int)triangle[0].y, fmin((int)triangle[1].y, (int)triangle[2].y));
+
+    //Finding the vertex lines between two points
+    Vertex vs1;
+    vs1.x = ((int)triangle[1].x - (int)triangle[0].x);
+    vs1.y = ((int)triangle[1].y - (int)triangle[0].y);
+    Vertex vs2;
+    vs2.x = (int)triangle[2].x - (int)triangle[0].x;
+    vs2.y = (int)triangle[2].y - (int)triangle[0].y;
+   
+   //Two for loops to go through both the y and the x 
+    for (int x = minX; x <= maxX; x++)
+    {
+        for (int y = minY; y <= maxY; y++)
+        {
+            //Our original vertex that we take the  x and y positing that we are at in the box
+            Vertex q;
+            q.x = (x - (int)triangle[0].x);
+            q.y = (y - (int)triangle[0].y);
+
+            //We find the cross product of each part and then divid them to find were we are if we are in the triangle
+            float s = (float)crossProduct(q, vs2) / crossProduct(vs1, vs2);
+            float t = (float)crossProduct(vs1, q) / crossProduct(vs1, vs2);
+
+            //If we are in the traingle these conditions will be true
+            if ( (s >= 0) && (t >= 0) && (s + t <= 1))
+            {
+                //set the point that we are at to send to the draw point
+                Vertex myPoint[1];
+                myPoint[0] = { x , y };
+                DrawPoint(target, myPoint, attrs, uniforms, frag);
```

Comment:
I simply set the target[y][x] value here.  But what you did in calling DrawPoint is fine.

Reply:
In fact, I forgot that there is a DrawPoint function to use, so good job in using your resources!